### PR TITLE
[feat](nereids) column prune on OlapScan

### PR DESCRIPTION
**Pull request by englefly**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/ColumnPruning.java [hunk 1]
```diff
@@ -141,6 +143,15 @@ public Plan visit(Plan plan, PruneContext context) {
         }
     }
 
+    @Override
+    public Plan visitLogicalOlapScan(LogicalOlapScan scan, PruneContext context) {
+        LogicalProperties origLogicalProperties = scan.getLogicalProperties();
+        LogicalProperties logicalProperties = new LogicalProperties(
+                () -> context.requiredSlots.stream().toList(),
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/ColumnPruning.java:150]: need compute `Sets.intersection(context.requiredSlots, scan.getOutput)`

**Comment by englefly** [comment]
run buildall

**Pull request closed without merging**